metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -327,6 +334,314 @@ def _convert_value(param: parameters.Parameter):
+)
+@click.option(
+    "--tag",
+    "tags",
+    multiple=True,
+    default=None,
+    help="Annotate all objects produced by Argo Workflows runs "
+    "with the given tag. You can specify this option multiple "
+    "times to attach multiple tags.",
+)
+@click.option(
+    "--sys-tag",
+    "sys_tags",
+    multiple=True,
+    default=None,
+    help="Annotate all Metaflow objects produced by Argo Metaflow runs "
+    "with the given system tag. You can specify this option multiple "
+    "times to attach multiple tags.",
+)
+@click.option(
+    "--namespace",
+    "user_namespace",
+    default=None,
+    help="Change the namespace from the default (production token) "
+    "to the given tag. See run --help for more information.",
+)
+@click.option(
+    "--k8s-namespace",
+    "--kubernetes-namespace",
+    "kubernetes_namespace",
+    default=KUBERNETES_NAMESPACE,
+    help="Kubernetes Namespace for your run in Argo.",
+    show_default=True,
+)
+@click.option(
+    "--yaml-only",
+    "yaml_only",
+    is_flag=True,
+    default=False,
+    help="Generate the Workflow YAML which is used to run the workflow on Argo.",
+    show_default=True,
+)
+@click.option(
+    "--pipeline-path",
+    "pipeline_path",
+    default=None,
+    help="The output path of the generated Argo pipeline yaml file",
+    show_default=False,
+)
+@click.option(
+    "--s3-code-package/--no-s3-code-package",
+    "s3_code_package",
+    default=True,
+    help="Whether to package the code to S3 datastore",
+    show_default=True,
+)
+@click.option(
+    "--base-image",
+    "base_image",
+    default=KFP_DEFAULT_CONTAINER_IMAGE,
+    help="Base docker image used in Argo.",
+    show_default=True,
+)
+@click.option(
+    "--max-parallelism",
+    "-m",
+    default=KFP_MAX_PARALLELISM,
+    show_default=True,
+    help="Maximum number of parallel pods.",
+)
+@click.option(
+    "--workflow-timeout", default=None, type=int, help="Workflow timeout in seconds."
+)
+@click.option(
+    "--notify",
+    "-n",
+    "notify",
+    is_flag=True,
+    default=bool(from_conf("METAFLOW_NOTIFY")),
+    help="Whether to notify upon completion.  Default is METAFLOW_NOTIFY env variable. "
+    "METAFLOW_NOTIFY_ON_SUCCESS and METAFLOW_NOTIFY_ON_ERROR env variables determine "
+    "whether a notification is sent.",
+    show_default=True,
+)
+@click.option(
+    "--notify-on-error",
+    "-noe",
+    "notify_on_error",
+    default=from_conf("METAFLOW_NOTIFY_ON_ERROR", default=None),
+    help="Email address to notify upon error. "
+    "If not set, METAFLOW_NOTIFY_ON_ERROR is used from Metaflow config or environment variable",
+    show_default=True,
+)
+@click.option(
+    "--notify-on-success",
+    "-nos",
+    "notify_on_success",
+    default=from_conf("METAFLOW_NOTIFY_ON_SUCCESS", default=None),
+    help="Email address to notify upon success"
+    "If not set, METAFLOW_NOTIFY_ON_SUCCESS is used from Metaflow config or environment variable",
+    show_default=True,
+)
```

Comment:
It looks like a lot of these options are duplicated from the `run` command. Can we reduce the duplications here? This SO question may be helpful: https://stackoverflow.com/questions/50061342/is-it-possible-to-reuse-python-click-option-decorators-for-multiple-commands